src/mds/SessionMap.h

```diff
@@ -661,6 +664,24 @@ class SessionMap : public SessionMapStore {
    */
   void save_if_dirty(const std::set<entity_name_t> &tgt_sessions,
                      MDSGatherBuilder *gather_bld);
+
+private:
+  void perflog_state(int state, bool entering);
+  void perflog_state_change(int old_state, int new_state);
+
+  class RequestLoadAvg {
+  public:
+    RequestLoadAvg(double half_life);
+    void adjust();
+    double get() const;
+  private:
+    DecayCounter dc;
+  };
+
+  std::map<entity_name_t, RequestLoadAvg> session_load_avg_map;
```

Comment:
You only need `DecayCounter` which is hit for any `SessionMap::hit_session` call. From it, you can calculate total load on the MDS:

> request load DecayCounter (hit for every request reply as requests may be retried)

Then you can also calculate average load per session by:

> avg request (request load DecayCounter / # of sessions)

i.e.

`SessionMap.total_load.get() / SessionMap.session_map.size()`

Reply:
Then, you can also add another `Session::DecayCounter` which is also hit. That one will not be used by the perf counters but we can still print it when dumping the sessions, see `MDSRank::dump_sessions`.

Reply:
> You only need DecayCounter which is hit for any SessionMap::hit_session call. From it, you can calculate total load on the MDS:

Got that. What I have in this PR for decay counters was mostly for testing and me trying to make sense of what the values correlate to. So, I had it per session so that I can kernel untar and see what's going on.